Mathlib/Analysis/Normed/Group/SeparationQuotient.lean

```diff
@@ -0,0 +1,591 @@
+      _ < ‖x‖ - ‖y - x‖ := by exact sub_lt_sub_left hy ‖x‖
+      _ = ‖x‖ - ‖x - y‖ := by rw [← norm_neg (y-x), ← neg_sub y x]
+      _ ≤ ‖x - (x - y)‖ := by exact norm_sub_norm_le x (x - y)
+      _ = ‖y‖ := by rw [sub_sub_self x y]
+    exact ne_of_gt (lt_of_le_of_lt (div_nonneg (norm_nonneg x) (zero_le_two)) this)
+
+instance : Nonempty (@nullSpace M) := ⟨0⟩
+
+/-- The definition of the norm on the quotient by an additive subgroup. -/
+noncomputable instance normOnSeparationQuotient : Norm (SeparationQuotient M) where
+  norm x := sInf (norm '' { m | mk m = x })
+
+theorem norm_eq (x : SeparationQuotient M) :
+    ‖x‖ = sInf (norm '' { m : M | mk m = x }) := rfl
+
+theorem norm_eq_infDist (x : SeparationQuotient M) :
+    ‖x‖ = infDist 0 { m : M | mk m = x } := by
+  simp only [norm_eq, infDist_eq_iInf, sInf_image', dist_zero_left]
+
+/-- An alternative definition of the norm on the quotient group: the norm of `mk x` is
+equal to the distance from `x` to `nullSpace`. -/
+theorem norm_mk (x : M) : ‖mk x‖ = infDist x (nullSpace).carrier := by
+  rw [norm_eq_infDist, ← infDist_image (IsometryEquiv.subLeft x).isometry,
+    IsometryEquiv.subLeft_apply, sub_zero, ← IsometryEquiv.preimage_symm]
+  congr 1 with y
+  simp only [mk_eq_mk, preimage_setOf_eq, IsometryEquiv.subLeft_symm_apply, mem_setOf_eq,
+    AddSubsemigroup.mem_carrier, AddSubmonoid.mem_toSubsemigroup, AddSubgroup.mem_toAddSubmonoid]
+  rw [inseparable_iff_norm_zero]
+  simp only [add_sub_cancel_right, norm_neg]
+  exact Eq.to_iff rfl
+
+theorem image_norm_nonempty (x : SeparationQuotient M) :
+    (norm '' { m | mk m = x }).Nonempty := .image _ <| Quot.exists_rep x
+
+theorem bddBelow_image_norm (s : Set M) : BddBelow (norm '' s) :=
+  ⟨0, forall_mem_image.2 fun _ _ ↦ norm_nonneg _⟩
+
+theorem isGLB_quotient_norm (x : SeparationQuotient M) :
+    IsGLB (norm '' { m | mk m = x }) (‖x‖) :=
+  isGLB_csInf (image_norm_nonempty x) (bddBelow_image_norm _)
+
+/-- The norm on the quotient satisfies `‖-x‖ = ‖x‖`. -/
+theorem quotient_norm_neg (x : SeparationQuotient M) : ‖-x‖ = ‖x‖ := by
+  simp only [SeparationQuotient.norm_eq]
+  congr 1 with r
+  constructor <;> { rintro ⟨m, hm, rfl⟩; use -m; simpa [neg_eq_iff_eq_neg] using hm }
+
+theorem quotient_norm_sub_rev (x y : SeparationQuotient M) : ‖x - y‖ = ‖y - x‖ := by
+  rw [← neg_sub, quotient_norm_neg]
+
+lemma norm_mk_eq_sInf (m : M) : ‖mk m‖ = sInf ((‖m + ·‖) '' nullSpace.carrier) := by
+  rw [norm_mk, sInf_image', ← infDist_image isometry_neg, image_neg]
+  have : -(@nullSpace M).carrier = nullSpace.carrier := by
+    ext x
+    rw [Set.mem_neg]
+    constructor
+    · intro hx
+      rw [← neg_neg x]
+      exact nullSpace.neg_mem' hx
+    · intro hx
+      exact nullSpace.neg_mem' hx
+  rw [this, infDist_eq_iInf]
+  simp only [dist_eq_norm', sub_neg_eq_add, add_comm]
+
+/-- The norm of the projection is equal to the norm of the original element. -/
+theorem quotient_norm_mk_eq (m : M) : ‖mk m‖ = ‖m‖ := by
```

Comment:
I think you should be taking this as the definition of the norm, and proving the other results about it.

Reply:
I think the current definition is parallel to the usual quotient norm (by the null space). Why should I take this as the definition in this case?

Reply:
I think a strong argument for doing so is to match the convention used to define `dist` here:

https://github.com/leanprover-community/mathlib4/blob/3ca1060d7e48260d1b1b7a0733f887e455455226/Mathlib/Topology/MetricSpace/Basic.lean#L191-L193

Reply:
I see, why is it that way, because you get some equations by definition?

I'm not sure how to keep it consistent between normed group and topological group for something else. For `InnerProductSpace`, I have to define a lift of `toDualMap`. Should I use the lift as just a linear map, continuous linear map or linear isometry? If linear isometry, should it be an extension of a continuous linear map?

Reply:
I've created #17452 with a version along the lines of what I was thinking. I think there's still a lot of good stuff in this PR, but it probably helps to build up the foundations a little more first.

Reply:
OK, I will wait until it will be merged. I guess I should lift `toDualMap` as a linear isometry?